app/server/routes/api.ts

```diff
@@ -88,45 +107,76 @@ router.post(
   "/validate/payment/dd",
   proxyApiHandler("https://payment." + conf.API_DOMAIN)(
     straightThroughBodyHandler
-  )("direct-debit/check-account", true)
+  )("direct-debit/check-account", "PAPI_VALIDATE_DIRECT_DEBIT", true)
 );
 
-router.use(
+router.post(
   "/case/:caseId?",
-  getAuthorisedExpressCallbackForApiGateway(
-    "cancellation-sf-cases-api",
-    pathParams => `/case/${pathParams.caseId || ""}`
-  )
+  cancellationSfCasesAPI("case", "CREATE_CANCELLATION_CASE")
+);
+router.patch(
+  "/case/:caseId?",
+  cancellationSfCasesAPI("case/:caseId", "UPDATE_CANCELLATION_CASE", "caseId")
 );
 
-router.use(
-  "/holidays/:subscriptionName?/:sfId?",
-  getAuthorisedExpressCallbackForApiGateway("holiday-stop-api", pathParams => {
-    const isPotentialCall = pathParams.sfId === "potential";
-    const basePathPart = isPotentialCall ? "potential" : "hsr";
-    const maybeActualExistingSfId = isPotentialCall
-      ? undefined
-      : pathParams.sfId;
-    return pathParams && pathParams.subscriptionName
-      ? `/${basePathPart}/${
-          pathParams.subscriptionName
-        }/${maybeActualExistingSfId || ""}`
-      : "/hsr";
-  })
+router.get(
+  "/holidays/:subscriptionName/potential",
+  holidayStopAPI(
+    "potential/:subscriptionName",
+    "HOLIDAY_STOP_POTENTIALS",
+    "subscriptionName"
+  )
+);
+router.get(
+  "/holidays/:subscriptionName",
+  holidayStopAPI(
+    "hsr/:subscriptionName",
+    "HOLIDAY_STOP_EXISTING",
+    "subscriptionName"
+  )
+);
+router.post("/holidays", holidayStopAPI("/hsr", "HOLIDAY_STOP_CREATE"));
+router.patch(
+  "/holidays/:subscriptionName/:sfId",
+  holidayStopAPI(
+    "hsr/:subscriptionName/:sfId",
+    "HOLIDAY_STOP_AMEND",
+    "subscriptionName",
+    "sfId"
+  )
+);
+router.delete(
+  "/holidays/:subscriptionName/:sfId",
+  holidayStopAPI(
+    "hsr/:subscriptionName/:sfId",
+    "HOLIDAY_STOP_WITHDRAW",
+    "subscriptionName",
+    "sfId"
+  )
 );
 
-router.use(
+router.get(
+  "/delivery-records/:subscriptionName",
+  deliveryRecordsAPI(
+    "delivery-records/:subscriptionName",
+    "DELIVERY_RECORDS_GET",
+    "subscriptionName"
+  )
+);
+router.post(
   "/delivery-records/:subscriptionName",
-  getAuthorisedExpressCallbackForApiGateway(
-    "delivery-records-api",
-    pathParams => `/delivery-records/${pathParams.subscriptionName}`
+  deliveryRecordsAPI(
+    "delivery-records/:subscriptionName",
+    "DELIVERY_PROBLEM_CREATE",
+    "subscriptionName"
```

Comment:
split these into the separate endpoints, so they could have unique `loggingCode`s
_also means these less 'logic' in the router (see https://github.com/guardian/manage-frontend/pull/351#discussion_r369083608)_